server/main.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const fcct = "./fcct-x86_64-unknown-linux-gnu"
+
+// Variables set via flag
+var (
+	// debug is if we should be printing out debug logs
+	debug bool
+	// host is the host to listen on
+	host string
+	// port is the port to listen on
+	port string
+)
+
+type postData struct {
+	ConfigString string `json:"config_string"`
+}
+
+type responseData struct {
+	Success        bool                   `json:"success"`
+	IgnitionConfig map[string]interface{} `json:"ignition_config"`
+	ErrLines       []int                  `json:"err_lines"`
+	ErrMsg         string                 `json:"err_msg"`
+}
+
+// write the response and sets the status
+func writeResponse(w *http.ResponseWriter, res responseData, status int) {
+	resJSON, err := json.Marshal(res)
+	if err != nil {
+		http.Error(*w, "Failed to parse struct `responseData` into JSON object", http.StatusInternalServerError)
+	}
+
+	(*w).Header().Set("Content-Type", "application/json")
+	(*w).WriteHeader(status)
+	(*w).Write(resJSON)
+}
+
+// runFCCT runs the FCCT command and returns the error, stdout, and stderr
+func runFCCT(input string) (string, string, error) {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd := exec.Command(fcct)
+	cmd.Stdin = strings.NewReader(input)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	err := cmd.Run()
+	return stdout.String(), stderr.String(), err
+}
+
+// logMiddleware produces simple logs when wrapped around a HandlerFunc
+func logMiddleware(handler http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		log.Printf("%s - %s - %s", r.RemoteAddr, r.Method, r.URL.Path)
+		if debug {
+			log.Printf("Request: %+v", r)
+		}
+		// Execute the original handler
+		handler(w, r)
+	}
+}
+
+// configHandler handles config requests when they come as a POST
+func configHandler(w http.ResponseWriter, r *http.Request) {
+	// enable CORS
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Methods", "POST, OPTIONS")
+	w.Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization")
+
+	// note: to enable CORS, it should handle OPTIONS as well
+	if r.Method == "OPTIONS" {
```

Comment:
Looks like OPTIONS ends up being blocked by nginx, but it's not causing a problem.

```
$ curl -X OPTIONS http://127.0.0.1:8007
<html>
<head><title>405 Not Allowed</title></head>
<body bgcolor="white">
<center><h1>405 Not Allowed</h1></center>
<hr><center>nginx/1.14.1</center>
</body>
</html>
```

Reply:
Scratch that, it works for `/config`